### PR TITLE
Remove unnecessary cast to Builder in generated Java lite code.

### LINKED ISSUE #5139: Generated Java code causes "redundant cast to Builder" warnings
**What version of protobuf and what language are you using?**
Version: v3.6.1
Gradle plugin version: v0.8.6
Language: Java

**What operating system (Linux, Windows, ...) and version?**
Ubuntu 16.04

**What runtime / compiler are you using (e.g., python version or gcc version)**
```
$ java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-0ubuntu0.16.04.1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
```

**What did you do?**

I don't know if this is a regression or always has been like this, this is the first time I'm using the protobuf java stuff.

I prepared a project demonstrating the issue https://github.com/slovdahl/protobuf-java-warnings, but in short:

1. Created a gradle project:
```gradle
buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.6'
    }
}

repositories {
    mavenCentral()
}

apply plugin: 'java'
apply plugin: 'com.google.protobuf'

protobuf {
    protoc {
        artifact = 'com.google.protobuf:protoc:3.6.1'
    }
}

dependencies {
    compile 'com.google.protobuf:protobuf-java:3.6.1'
}

compileJava {
    options.compilerArgs += ['-Xlint']
}
```
2. Created a proto definition:
```protobuf
syntax = "proto3";

package com.foo.bar.proto;

option java_outer_classname = "Event";

message Foo {
    int32 id = 1;
}
```
3. Executed `./gradlew compileJava`, which generates the java code and compiles the generated code.

**What did you expect to see**
No java compiler warnings.

**What did you see instead?**
```
/home/dev/protobuf-test/build/generated/source/proto/main/java/com/foo/bar/proto/Event.java:341: warning: [cast] redundant cast to Builder
        return (Builder) super.clone();
               ^
/home/dev/protobuf-test/build/generated/source/proto/main/java/com/foo/bar/proto/Event.java:347: warning: [cast] redundant cast to Builder
        return (Builder) super.setField(field, value);
               ^
/home/dev/protobuf-test/build/generated/source/proto/main/java/com/foo/bar/proto/Event.java:352: warning: [cast] redundant cast to Builder
        return (Builder) super.clearField(field);
               ^
/home/dev/protobuf-test/build/generated/source/proto/main/java/com/foo/bar/proto/Event.java:357: warning: [cast] redundant cast to Builder
        return (Builder) super.clearOneof(oneof);
               ^
/home/dev/protobuf-test/build/generated/source/proto/main/java/com/foo/bar/proto/Event.java:363: warning: [cast] redundant cast to Builder
        return (Builder) super.setRepeatedField(field, index, value);
               ^
/home/dev/protobuf-test/build/generated/source/proto/main/java/com/foo/bar/proto/Event.java:369: warning: [cast] redundant cast to Builder
        return (Builder) super.addRepeatedField(field, value);
               ^
6 warnings
```

### PR DESCRIPTION
**Pull request by 1e0ng**
Fixes #5139 

Previously the fix https://github.com/protocolbuffers/protobuf/pull/5247 has removed unnecessary cast to Builder in generated Java code, and the current PR is doing a similar one for Java lite code.

After this PR, both Java and Java lite code won't have an unnecessary cast to Builder, thus #5139 will be fixed.

This change simply removed those casts to Builder for 2 methods, and the reason it will work is:

DEFAULT_INSTANCE.createBuilder method is defined in the superclass, and the return value is a generic value which is exactly the Builder class passed in.

### CODE REVIEW

**Comment by 1e0ng** [comment]
@googleberg Hi Jerry, could you help take a look at this PR?

**Comment by googleberg** [comment]
Hello @1e0ng, please sign the CLA and then I'll be happy to take a look.

**Comment by 1e0ng** [comment]
> Hello @1e0ng, please sign the CLA and then I'll be happy to take a look.

Yes, I have signed the CLA. I think it's the Mergeable check that fails but not sure why it mentions CLA check though.

**Review by googleberg — APPROVED** [review]

**Comment by 1e0ng** [comment]
Thanks for taking a look, @googleberg I have rebased my PR. Could you trigger the CI again?

**Comment by 1e0ng** [comment]
It looks like the label `java-lite` is not an acceptable label. We might need to either add `java-lite` into the language list or just use `java` tag.

**Comment by 1e0ng** [comment]
@googleberg could you help approve running workflows again?

**Comment by googleberg** [comment]
running

**Comment by 1e0ng** [comment]
All the checks have passed now except one called "Mergeable", I'm out of clue why it fails now.

**Comment by googleberg** [comment]
I think we are in good shape. We have an automated process to merge now and I think this is to prevent me from accidentally merging manually.

**Comment by 1e0ng** [comment]
I see why now. It shows `At least 2 approving reviews are required by reviewers with write access.` Could you get one more approval from a reviewer with write access?

**Review by shaod2 — APPROVED** [review]

**Review by shaod2 — APPROVED** [review]

**Comment by 1e0ng** [comment]
Hi @haberman PR rebased.

**Comment by fowles** [comment]
kicked the workflow, it looks like you are synced to a bad point.  Can you rebase again?  (sorry)

**Comment by 1e0ng** [comment]
> kicked the workflow, it looks like you are synced to a bad point. Can you rebase again? (sorry)

Ok, rebased again. Hopefully this time it works.

**Comment by 1e0ng** [comment]
Hi @haberman did it work this time?

**Review by fowles — APPROVED** [review]

**Pull request closed without merging**
